lib/carrierwave-aws.rb

```diff
@@ -43,7 +44,33 @@ def self.normalized_acl(acl)
     normalized
   end
 
+  def self.aws_signer(value = nil)
+    self.aws_signer = value if value
+
+    if instance_variable_defined?('@aws_signer')
+      @aws_signer
+    elsif superclass.respond_to? :aws_signer
+      superclass.aws_signer
+    end
+  end
+
+  def self.aws_signer=(signer)
+    @aws_signer = validated_signer(signer)
+  end
+
+  def self.validated_signer(signer)
+    unless signer.nil? || signer.instance_of?(Proc) && signer.arity == 2
```

Comment:
@sorentwo:

`signer.nil?` was added to allow `uploader.aws_acl = nil` or `CertainUploader.aws_acl = nil` in case of `CarrierWave::Uploader::Base.aws_acl` is already set but should not be used in a derived class or a certain instance.

For the second part: 
Checking the signer to respond to call is what I had in mind first, but how to check the arity of the call method? I'd like to enforce the interface 'call' with the two parameters. There seems no easy way to do that for a proc or a class- or instance-method the same way. Therefor the signer is forced to be a proc. Do you have a better idea doing this?